packages/query-store/src/QueryStore.ts

```diff
@@ -93,6 +102,32 @@ export class QueryStore extends AbstractStore<QueryStoreValue> {
 
 	readonly id: string;
 
+	#error: Error | unknown;
+
+	#setError = (e: Error | unknown): void => {
+		console.debug(`QueryStore encountered a non-fatal error`, e);
+		this.#error = e;
+
+		if (e) {
+			// This needs some testing; might cause a race, but we want to prevent the query from loading anything at this point.
+			this.#length = 0;
+			this.#lengthLoaded = false;
+			this.#lengthLoading = false;
+		}
+
+		if (this.opts.errorNotifier) this.opts.errorNotifier(this.error!);
+
+		this.publish();
+	};
```

Comment:
```suggestion
	#setError = (err: Error | unknown): void => {
		console.debug(`QueryStore encountered a non-fatal error`, err);
		this.#error = err;

		if (err) {
			// This needs some testing; might cause a race, but we want to prevent the query from loading anything at this point.
			this.#length = 0;
			this.#lengthLoaded = false;
			this.#lengthLoading = false;
		}

		if (this.opts.errorNotifier) this.opts.errorNotifier(this.error!);

		this.publish();
	};
```